examples/echo/main.c

```diff
@@ -143,16 +144,35 @@ static void handle_rx (YggWorker   *worker,
   g_bytes_unref (data);
 }
 
+static gboolean
+sigint_callback(gpointer main_loop) {
+  // Hint: this is good place to perform any resource clean up
+  // before the application terminates.
+  g_debug ("handling signal SIGINT");
+  // Stops main loop
+  g_main_loop_quit(main_loop);
```

Comment:
This call should have a space as well.
```suggestion
  g_main_loop_quit (main_loop);
```
It is relatively safe to assume that the data passed into the callback function will be an instance of `GMainLoop`, but since this is demonstrative code, I'd rather we do something that checks for type safety first.
```suggestion
  static gboolean
  sigint_callback (gpointer user_data)
  [...]
  GMainLoop *main_loop = G_MAIN_LOOP (user_data);
  g_main_loop_quit(main_loop);
```
This will raise an assertion error in the event `user_data` cannot be cast safely as a `GMainLoop`.